java/src/main/java/org/whispersystems/signalservice/api/SignalServiceMessageSender.java

```diff
@@ -170,18 +170,30 @@ public void sendMessage(SignalServiceAddress recipient, SignalServiceDataMessage
   public void sendMessage(List<SignalServiceAddress> recipients, SignalServiceDataMessage message)
       throws IOException, EncapsulatedExceptions
   {
-    byte[]              content   = createMessageContent(message);
-    long                timestamp = message.getTimestamp();
-    SendMessageResponse response  = sendMessage(recipients, timestamp, content, true);
+    byte[]                  content                 = createMessageContent(message);
+    long                    timestamp               = message.getTimestamp();
+    EncapsulatedExceptions  encapsulatedExceptions  = null;
+
+    try {
+      sendMessage(recipients, timestamp, content, true);
+    } catch (EncapsulatedExceptions e) {
+      encapsulatedExceptions = e;
+    }
 
     try {
-      if (response != null && response.getNeedsSync()) {
+      if (!store.getSubDeviceSessions(localAddress.getNumber()).isEmpty()) {
```

Comment:
They must by synced on failure, because otherwise they will never be synced when you have an unregistered number in your group, which is exactly what i am fixing here.

I did not yet change it in PushServiceSocket because Signal-Android apparently does not use it and i did not notice that it is wrong there too.